hid-replay.py

```diff
@@ -32,36 +33,31 @@ def __init__(self, filename):
         self.filename = filename
         with open(filename) as f:
             idx = 0
-            name = None
-            rdesc = None
-            phys = None
-            info = None
             for l in f.readlines():
+                l = l.strip()
                 if l.startswith('D:'):
-                    idx = int(l.split(' ')[1])
-                    name = None
-                    rdesc = None
-                    phys = None
-                    info = None
+                    r = parse('D: {idx:d}', l)
+                    assert r is not None
+                    idx = r['idx']
                     continue
                 if idx not in self._devices:
                     self._devices[idx] = uhid.UHIDDevice()
                 dev = self._devices[idx]
                 if l.startswith('N:'):
-                    name = l.split(' ', 1)[1].strip()
-                    dev.name = name
+                    r = parse('N: {name}', l)
+                    assert r is not None
+                    dev.name = r['name']
                 elif l.startswith('I:'):
-                    bus, vid, pid = l.split(' ')[1:]
-                    info = [int(i, 16) for i in (bus, vid, pid)]
-                    dev.info = info
+                    r = parse('I: {bus:x} {vid:x} {pid:x}', l)
+                    assert r is not None
+                    dev.info = [r['bus'], r['vid'], r['pid']]
                 elif l.startswith('P:'):
-                    phys = l.split(' ', 1)[1].strip()
-                    dev.phys = phys
+                    r = parse('P: {phys}', l)
+                    if r is not None:
+                        dev.phys = r['phys']
                 elif l.startswith('R:'):
-                    length, rdesc = l.split(' ', 2)[1:]
-                    rdesc = [int(r, 16) for r in rdesc.split()]
-                    assert int(length) == len(rdesc)
-                    dev.rdesc = rdesc
+                    r = findall(' {:x}', l)
```

Comment:
The first item in the list is the length of the rdesc. It should be parsed as a plain decimal and should be stripped out of the report descriptor.